### PR TITLE
Kernel+Userland: Add basic SysFS /sys/devices/storage/ directory for exposing Storage Devices, Introduce the lsblk utility

### PR DESCRIPTION
**Pull request by supercomputer7**
Heavily relies on #13772. It works quite nicely with the changes to #13772 so I'm pretty happy about the overall changes.
Of course, this is just another step towards writing proper udev-like program for Serenity. We still needto adapt more devices to use this pattern of a symlink between `/sys/dev/{block|char}/{link_to_actual_device}` and a device tree under `/sys/devices/` directory. The storage device is just the first object to use this :)

### CODE REVIEW

**Comment by supercomputer7** [comment]
Putting this as a side note because I think it might be interesting for us later on - the way we use LUNs is quite nice in this PR, but I expect this to propagate further on, so for example, we could use LUNs to specify the boot drive instead of using the weird and not-reflective `/dev/hdaX` pattern, which really doesn't tell anything about the actual device we specified nor there's a guarantee that the same `/dev/hdaX` device node on `DevTmpFS` will be the same.

**Comment by supercomputer7** [comment]
This is not stale.

File: Kernel/FileSystem/SysFS/Subsystems/DeviceIdentifiers/SymlinkDeviceComponent.h [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+
+public:
+    static NonnullRefPtr<SysFSSymlinkDeviceComponent> must_create(SysFSDeviceIdentifiersDirectory const& parent_directory, Device const&, SysFSComponent const& pointed_component);
+    virtual StringView name() const override { return m_major_minor_formatted_device_name->view(); }
+    bool is_block_device() const { return m_block_device; }
+
+private:
+    SysFSSymlinkDeviceComponent(SysFSDeviceIdentifiersDirectory const& parent_directory, NonnullOwnPtr<KString> major_minor_formatted_device_name, Device const&, SysFSComponent const& pointed_component);
+    IntrusiveListNode<SysFSSymlinkDeviceComponent, NonnullRefPtr<SysFSSymlinkDeviceComponent>> m_list_node;
+    bool m_block_device;
```
Comment (supercomputer7) [inline Kernel/FileSystem/SysFS/Subsystems/DeviceIdentifiers/SymlinkDeviceComponent.h:29]: I should probably mark this as `const` and use a default-value here.

**Comment by awesomekling** [comment]
1. What is a "soft link"?

2. Commit messages like "Add two methods related to relative paths for components" and "Add two protected methods for DeviceManagement functions" are not very helpful, since they don't give any clue about what is being added and why. I can look at the diff for 2 seconds and determine that two functions are being added, but it will take much longer to understand what they are doing. :)

**Comment by awesomekling** [comment]
Oh, and this has conflicts

**Comment by supercomputer7** [comment]
> 1. What is a "soft link"?
> 
>     2. Commit messages like "Add two methods related to relative paths for components" and "Add two protected methods for DeviceManagement functions" are not very helpful, since they don't give any clue about what is being added and why. I can look at the diff for 2 seconds and determine that two functions are being added, but it will take much longer to understand what they are doing. :)

1. Soft link as symlink, I should have been more clear about this. Still, this is a soft "link" that the kernel generates in the SysFS, so maybe that's why I choose a different word for this.
2. I agree, I'll fix this and send a fix :)

**Pull request merged**